### ISSUE TITLE
Build process get stuck on build neqo-crypto

### ISSUE THREAD

**Issue by CrashingBrain** [mozilla/neqo#1016]
after installing all dependencies, trying to build using `cargo build` goes without errors until neqo-crypto (120/129), then it gets stuck there. I let it run for more than an hour so I assumed it's stuck. 

I am working on macOS Mojave 10.14.6 and I followed the steps from #208 
```
brew install python3 rust nss
git clone https://chromium.googlesource.com/external/gyp
cd gyp
python3 setup.py install
cd ..
git clone git@github.com:mozilla/neqo.git 
cd neqo
cargo build
```

So I think I am ok with gyp and nss. Anyway here is what I have
```
Name: gyp
Version: 0.1
Summary: Generate Your Projects
Home-page: http://code.google.com/p/gyp
Author: Chromium Authors
Author-email: chromium-dev@googlegroups.com
License: UNKNOWN
Location: /Users/luca/.pyenv/versions/3.8.1/lib/python3.8/site-packages/gyp-0.1-py3.8.egg
```
```
nss: stable 3.58 (bottled)
nspr: stable 4.29 (bottled)
```

Unfortunately there is no error message I can show.

**Comment by martinthomson** [comment]
It sounds like something is hanging waiting for input.  Does `ps` give you any clues?

**Comment by CrashingBrain** [comment]
I don't really know how to read this

```
  PID TTY           TIME CMD
 1070 ttys000    0:00.01 -bash
 1243 ttys001    0:00.01 -bash
 1332 ttys001    0:00.15 /Users/luca/.rustup/toolchains/stable-x86_64-apple-darwin/bin/cargo build
 1335 ttys001    0:00.00 /Users/luca/neqo/target/debug/build/neqo-crypto-ec1b400f094b6707/build-script-build
 1336 ttys001    0:00.01 bash ./build.sh -Ddisable_tests=1 --static
 1390 ttys001    0:00.31 /Users/luca/.pyenv/versions/3.8.1/bin/python /Users/luca/.pyenv/versions/3.8.1/bin/gyp -f ninja --depth=/Users/luca/neqo/target/debug
 1406 ttys001    0:00.03 /Users/luca/.pyenv/versions/3.8.1/bin/python -c from multiprocessing.resource_tracker import main;main(8)
 2514 ttys001    0:00.13 /Users/luca/.pyenv/versions/3.8.1/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=9, pipe_handle=20
 2515 ttys001    0:00.13 /Users/luca/.pyenv/versions/3.8.1/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=9, pipe_handle=22
 2224 ttys002    0:00.01 -bash
```

It seems that it is working though. I tried at some time intervals and it didn't change

**Comment by martinthomson** [comment]
That looks like gyp is running.

FWIW, fetching and building NSS can take an awful long time, and that can make this appear to be stalling, particularly if you have slow network access.  My NSS checkout is 800M.  And neqo currently depends on NSS source being available.  I'd welcome a patch to change that; it's fairly simple to cut out building of NSS and put that behind a feature flag.  Failing that, if you follow the instructions for getting a separate copy of NSS, that can help quite a bit, especially if you ever have to `cargo clean`.

**Comment by CrashingBrain** [comment]
I don't know what you mean with "awful long time", but I did another test for more hours to no result.
For the moment, I actually went another way and installed neqo through a docker image and there it finished the build process in about 15 minutes. So I would exclude the network problems.

I tried the instructions for getting a separate copy of NSS and building it from source in the repo, and from there I saw that it gets stuck at the stage of installing NSPR. Don't know if that's helpful to you.

I'll keep trying to look into this to see if I can manage the NSS build/installation but meanwhile I am working on docker.
I'll update if I find a solution
